### DISCUSSION TITLE
SOLVED: ignore_errors not working?

### DISCUSSION THREAD

**Discussion by jimmywan** [tox-dev/tox#2137] [Q&A] [answered]
I'm trying to setup mypy on a project that doesn't have fully correct type hints, but since it's already working, I didn't want to block builds on mypy errors and to just treat it as an informational nice-to-have.
I annotated my tox.ini as follows:
```
[testenv:mypy]
ignore_errors = true
commands =
    poetry run mypy app
```

but tox.ini still returns a non-zero error code.

```bash
(tox -e mypy && echo "success")|| echo "failed"
...
ERROR:   mypy: commands failed
failed
```

Am I doing something wrong or is this a tox bug?

Using 3.23.1.

**Comment by gaborbernat** [comment, marked as answer]
You want the - prefix per https://tox.readthedocs.io/en/latest/config.html#conf-commands, ignore errors only allows you to continue but keeps the exit code see https://tox.readthedocs.io/en/latest/config.html#conf-ignore_errors

**Reply by jimmywan** [reply]
Awesome, ty!

**Reply by jugmac00** [reply]
I type too slow on mobile 🤣 👍

**Comment by jugmac00** [comment]
The option is documented here
https://tox.readthedocs.io/en/latest/config.html#conf-ignore_errors

It just means that if there is more than one command and the first fails, continue to run the others, but still return non-zero.

You want a non-zero return code, altough you have a failing command. Just prepend the command with a dash, e.g. `commands = - mypy`, see
https://tox.readthedocs.io/en/latest/config.html#conf-commands
